tests/Services/GetPreliminaryInvoiceTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Tests\Services;
+
+use App\Services\GetMeteringData;
+use App\Services\GetPreliminaryInvoice;
+use App\Services\GetSpotPrices;
+use Mockery\MockInterface;
+use Tests\TestCase;
+
+class GetPreliminaryInvoiceTest extends TestCase
+{
+
+    private $charges;
+    private mixed $spotPrices;
+    private mixed $testConsumptions;
+
+
+    protected function setUp(): void
+    {
+
+        //Got hold of some real charges and decided to use them here - not so important anyway what the values are
+        //but we need a reliable datastructure.
+        $this->charges = $this->loadTestData(fixture_path('typical_charges.json'));
+
+        $this->spotPrices = $this->loadTestData(fixture_path('spot_prices.json'));
+
+        //Consumption is normally returned from the service as an array with {date_time=>usage}
+        //So a key with 00:00 means the usage between 00:00 and 01:00
+        $this->testConsumptions = $this->loadTestData(fixture_path('consumption_data.json'));
+
+
+        parent::setUp();
+    }
+
+    /**
+     * Goal here is to check that the calculations can be performed on the expected datastructes.
+     * Won't test the actual data/data-structures from providers - this is should be done in
+     * other test (and is also important)
+     *
+     * @throws \App\Exceptions\DataUnavailableException
+     * @throws \Tvup\ElOverblikApi\ElOverblikApiException
+     * @throws \Tvup\EwiiApi\EwiiApiException
+     */
+    public function testGetBill()
```

Comment:
Jeg er kæmpe fan af det her  - Det her er et eksempel på hvor Mock er fucking nice! .. jeg er selv sygt dårlig til at bruge det nemlig .. så mega fedt! 👍🏼

Reply:
Ved mindre folk går fuldstændig, mock, mock, mock alt, så bliver det nederen. Jeg synes, det er fedt med mock at fjerne fokus fra det, som ikke skal testes i den kontekst man er i. Og så skal man nok sørge for at have den rigtige balance, så man ikke mocker applikationen ihjel og ikke får en sikkerhed for, at den kan tale sammen med resten af verden ud udføre en opgave end-2-end.
Men jeg er nu ikke så super stærk i disciplinen :)